src/copyright/agent/database.cc

```diff
@@ -300,21 +300,38 @@ bool CopyrightDatabaseHandler::insertNoResultInDatabase(long int agentId, long i
 
 bool CopyrightDatabaseHandler::insertInDatabase(DatabaseEntry& entry) const
 {
-  return dbManager.execPrepared(
-    fo_dbManager_PrepareStamement(
-      dbManager.getStruct_dbManager(),
-      "insertInDatabase",
-      "INSERT INTO "
-      IDENTITY
-      "(agent_fk, pfile_fk, content, hash, type, copy_startbyte, copy_endbyte)"
-        " VALUES($1,$2,$3,md5($3),$4,$5,$6)",
-      long, long, char*, char*, int, int
-    ),
-    entry.agent_fk, entry.pfile_fk,
-    entry.content.c_str(),
-    entry.type.c_str(),
-    entry.copy_startbyte, entry.copy_endbyte
-  );
+  if(!strcmp(entry.type.c_str(),IDENTITY) || strcmp(entry.type.c_str(),"statement")){
```

Comment:
>> The only other option for IDENTITY is "ecc" (see identity.hpp). This implies that only ecc is placed in another table called ecc and everything else is placed in the table called author? At least this is how I understand the code.

Here i'm comparing the IDENTITY with "entry.type.c_str()" which doesn't contain "copyright". More over there are almost 4 types "ecc, statement, email, url, author" that's why it would be better to compare it with IDENTITY.